dot/peerset/handler.go

```diff
@@ -30,67 +34,89 @@ func NewPeerSetHandler(cfg *ConfigSet) (*Handler, error) {
 	}, nil
 }
 
+func (h *Handler) setActionQueue(act action) {
+	go func(data action) {
+		h.writerWGMutex.Lock()
+		h.writersWG.Add(1)
+		h.writerWGMutex.Unlock()
```

Comment:
You should add to the writes wait group before launching the goroutine. And also the mutex protecting the waitgroup looks sketchy, is it really needed?